pkg/machinehealthcheck/machinehealthcheck.go

```diff
@@ -118,7 +115,51 @@ var _ = Describe("[Feature:MachineHealthCheck] MachineHealthCheck", func() {
 		Expect(err).ToNot(HaveOccurred())
 	})
 
-	// It("should respect maxUnhealthy", func() {
-	// 	// TODO(bison): This.
-	// })
+	It("should respect maxUnhealthy", func() {
+		By("Setting unhealthy conditions on machine nodes")
+
+		selector := machineSet.Spec.Selector
+		machines, err := framework.GetMachines(client, &selector)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(machines).ToNot(BeEmpty())
+
+		for _, machine := range machines {
+			node, err := framework.GetNodeForMachine(client, machine)
+			Expect(err).ToNot(HaveOccurred())
+			err = framework.AddNodeCondition(client, node, nodeCondition)
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		By("Creating a MachineHealthCheck resource with low maxUnhealthy threshold")
```

Comment:
what is low? can we be explicit?

Reply:
can we do something more disruptive with a higher value for maxUnhealthy? wait for a few machines to be remediated and see the MHC short circuit for the others?

Reply:
I extended the case and made it more visible, so it will ensures the machines will be remediated when the number of unhealthy entities does not reach the threshold, and won't be past that. Also added checks for all events being emitted during that procedure.